src/Juvix/Compiler/Concrete/Translation/FromSource.hs

```diff
@@ -564,11 +564,20 @@ parsePatternAtoms = do
 
 functionClause :: Members '[InfoTableBuilder, JudocStash, NameIdGen] r => Symbol -> ParsecS r (FunctionClause 'Parsed)
 functionClause _clauseOwnerFunction = do
-  _clausePatterns <- P.many patternAtom
-  kw kwAssign
+  _clausePatterns <- functionClausePatterns
   _clauseBody <- parseExpressionAtoms
   return FunctionClause {..}
 
+functionClausePatterns :: forall r. Members '[InfoTableBuilder, JudocStash, NameIdGen] r => ParsecS r [PatternAtom 'Parsed]
+functionClausePatterns =
+  P.manyTill patternAtom (kw kwAssign <|> wrongEq)
+  where
+    wrongEq :: ParsecS r ()
+    wrongEq = do
+      P.chunk "="
```

Comment:
this will also rule out `==` or any identifier starting with `=`. Was this intended?

Reply:
I fixed this. Now only '=' exactly causes an error.